report/renderer/util.js

```diff
@@ -138,6 +138,10 @@ export class Util {
           });
         }
       });
+
+      category.auditRefs.sort((a, b) => {
```

Comment:
this might be a better fit in the category renderer since we're really using it for rendering side effects. That might(?) also allow easier splitting on category type, since it seems like we shouldn't be doing this for the perf category, at least

Reply:
hadn't realized it changed the order of the metrics in the metric table. But the new order seems fine? is there any intention behind the current order?

> this might be a better fit in the category renderer

https://github.com/GoogleChrome/lighthouse/blob/344a1560824e658792fe0402fe090a0f56b44d0d/report/renderer/category-renderer.js#L516

make a new array here `const auditRefsByWeight = [...auditRefs].sort(...)`, or mutate the array (mutating was the only thing I had considered, which is why I thought to do it in the only place we mutate the LHR)?

Reply:
> make a new array here `const auditRefsByWeight = [...auditRefs].sort(...)`,

this one sounds good to me

Reply:
> hadn't realized it changed the order of the metrics in the metric table. But the new order seems fine? is there any intention behind the current order?
> 
> >

I believe there is an intentional order (or maybe there _was_ an intentional order and we've organically evolved since then?), but we also don't necessarily want to rearrange things on people (at least in a minor release) since you get pretty used to the metric order

Reply:
v9 then?

at the moment, LCP isn't even next to FCP... + things aren't sorted by weight ... so no idea if there is any logical framework to the order right now.

Reply:
I think it may have been more like early -> mid -> late load metrics? @paulirish

Reply:
> I think it may have been more like early -> mid -> late load metrics? @paulirish

yup. that's been the logic to the metric order. 

also FCP/SI/TTI have been in the same location since v5 (and perhaps even earlier).

at least in the metric case, we have a link about how it works plus the calculator, so i dont feel bad if we dont have the metrics sorted by weight.

Reply:
> this might be a better fit in the category renderer since we're really using it for rendering side effects.

i dont care too much about placement.. this location also seems fine.
but i do think we should skip performance category entirely. (really only to avoid moving the metrics... but we know the rest of the category is all 0 weight)

Reply:
(aside from my above comments.. i think this definitely improves the accessibility category) 👍

Reply:
>  connorjclark added the 9.0 label yesterday

If we don't change the perf audit ordering (which I don't think we should do anyways), I think it'd be fine to ship sorting the accessibility, etc audits without a breaking change

Reply:
> i dont care too much about placement.. this location also seems fine.

I really think it should be in category-renderer:

It's category rendering, not preparing data for compatibility. If you go looking for what's ordering audits in a category in the report, the most logical place is in `CategoryRenderer.render()` (or LHR generation, but agreed with https://github.com/GoogleChrome/lighthouse/pull/13053#issuecomment-919389046). It's easy to unit test that way. We get abstaining from performance category renderer changes automatically. win-win-win-win :)

Reply:
And it would parallel the equivalent functionality in performanceCategoryRenderer for opportunities 

https://github.com/GoogleChrome/lighthouse/blob/1e225484aefd793c200fb979fe73822fe218d88d/report/renderer/performance-category-renderer.js#L214-L216

and diagnostics

https://github.com/GoogleChrome/lighthouse/blob/1e225484aefd793c200fb979fe73822fe218d88d/report/renderer/performance-category-renderer.js#L247-L253

Reply:
I agree, I just haven't gotten around to it yet.